dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/AbstractSummaryHolderJdbcConnector.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2022 Google LLC
+ * Copyright 2013-2021 CompilerWorks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.edwmigration.dumper.application.dumper.connector;
+
+import java.time.ZonedDateTime;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import javax.annotation.Nonnull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author shevek
+ */
+public abstract class AbstractSummaryHolderJdbcConnector extends AbstractJdbcConnector {
+
+  @SuppressWarnings("UnusedVariable")
+  private static final Logger LOG = LoggerFactory.getLogger(
+      AbstractSummaryHolderJdbcConnector.class);
+
+  private final Map<String, Summary> metadataMap;
+
+  public AbstractSummaryHolderJdbcConnector(@Nonnull String name) {
+    super(name);
+    metadataMap = new ConcurrentHashMap<>();
+  }
+
+  public Summary getSummary() {
+    ZonedInterval interval = new ZonedInterval(
+        getEarliestQueryTime(),
+        getLatestQueryTime()
+    );
+    return new Summary(getTotalRows(), interval);
+  }
+
+  protected void updateMetadata(String queryType, long rowCount, ZonedInterval interval) {
+    metadataMap.putIfAbsent(queryType, new Summary(rowCount, interval));
+    metadataMap.computeIfPresent(queryType,
+        (q, metadata) -> metadata.updateMetadata(rowCount, interval));
```

Comment:
If `rowCount` passed as a parameter to this `updateMetadata` method is 1, then what would be the value of `Summary#rowCount` field after the `putIfAbsent` and `computeIfPresent` invocations?

Reply:
Nice catch! Thanks